File: SDKs/motaword.com/1.0/go/pkg/models/operations/updateproject.go
package operations

import (
	"openapi/pkg/models/shared"
)

type UpdateProjectPathParams struct {
	ID int64 `pathParam:"style=simple,explode=false,name=id"`
}

type UpdateProjectRequestBody struct {
	CallbackURL     *string  `multipartForm:"name=callback_url"`
	CouponCode      *string  `multipartForm:"name=coupon_code"`
	Custom          []string `multipartForm:"name=custom,json"`
	SourceLanguage  *string  `multipartForm:"name=source_language"`
	TargetLanguages []string `multipartForm:"name=target_languages[],json"`
}

type UpdateProjectRequest struct {
	PathParams UpdateProjectPathParams
	Request    *UpdateProjectRequestBody `request:"mediaType=multipart/form-data"`
}

type UpdateProjectResponse struct {
	ContentType string
	Error       *shared.Error
	Project     *shared.Project
	StatusCode  int64
}

File: SDKs/asana.com/1.0/python/src/sdk/models/operations/updateproject.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import projectrequest as shared_projectrequest
from ..shared import projectresponse as shared_projectresponse
from ..shared import errorresponse as shared_errorresponse


@dataclasses.dataclass
class UpdateProjectPathParams:
    project_gid: str = dataclasses.field(metadata={'path_param': { 'field_name': 'project_gid', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class UpdateProjectQueryParams:
    opt_fields: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'opt_fields', 'style': 'form', 'explode': False }})
    opt_pretty: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'opt_pretty', 'style': 'form', 'explode': True }})
    

@dataclass_json
@dataclasses.dataclass
class UpdateProjectRequestBodyInput:
    data: Optional[shared_projectrequest.ProjectRequestInput] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class UpdateProject200ApplicationJSON:
    data: Optional[shared_projectresponse.ProjectResponse] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclasses.dataclass
class UpdateProjectRequest:
    path_params: UpdateProjectPathParams = dataclasses.field()
    query_params: UpdateProjectQueryParams = dataclasses.field()
    request: UpdateProjectRequestBodyInput = dataclasses.field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class UpdateProjectResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    error_response: Optional[shared_errorresponse.ErrorResponse] = dataclasses.field(default=None)
    update_project_200_application_json_object: Optional[UpdateProject200ApplicationJSON] = dataclasses.field(default=None)
    
